feat(executor/grpc): Add GRPC executor (ovh#187)

matoous opened the pull request:
Add GRPC executor which uses grpcurl to do requests to grpc endpoints.

Fixes: https://github.com/ovh/venom/issues/187
Signed-off-by: Matous Dzivjak <matous.dzivjak@kiwi.com>

matoous commented on executors/grpc/executor.go:
```diff
@@ -0,0 +1,222 @@
+	if e.Service != "" {
+		args = append(args, fmt.Sprintf("%s/%s", e.Service, e.Method))
+	} else {
+		args = append(args, fmt.Sprintf("%s", e.Method))
```
Fixed

matoous commented on executors/grpc/executor.go:
```diff
@@ -0,0 +1,222 @@
+	result.Systemout = systemOutBuilder.String()
+	result.Systemerr = systemErrBuilder.String()
+	result.TimeSeconds = elapsed.Seconds()
+	result.TimeHuman = fmt.Sprintf("%s", elapsed)
```
Fixed

yesnault commented on executors/grpc/README.md:
````diff
@@ -0,0 +1,81 @@
+## Default assertion
+
+```yaml
+result.code ShouldEqual 0
````
It's result.statuscode, no?

matoous replied:
Indeed, fixed, thanks.

matoous commented:
@yesnault fixed.

yesnault commented:
@matoous thank you! can't wait to test it :)
Missing a little impact on https://github.com/ovh/venom/blob/master/README.md#executors to add the new GRPC executor

matoous commented:
Added. Also fixed the codes returned by grpc executer.

mrVanboy commented:
The things I believe should be mentioned:
- this configuration relies on the gRPC server reflection, which should be enabled on the server: https://github.com/grpc/grpc/blob/master/doc/server-reflection.md
- gRPC server reflection is not enabled by default and not implemented for every gRPC library
- gRPC server reflection does not properly work with `gogo/protobuf`: https://github.com/grpc/grpc-go/issues/1873

matoous commented:
Thanks @mrVanboy, I added all of your points to README. Also added a little bit more about the return codes. Added more configuration and also more comments to the code. Should be ready now.

yesnault commented on executors/grpc/README.md:
```diff
@@ -0,0 +1,83 @@
+    service: coolService.api
+    method: GetAllFoos
+    assertions:
+    - result.statuscode ShouldEqual 0
```
```suggestion
    - result.code ShouldEqual 0
```

matoous replied:
Fixed, thanks!

yesnault commented on tests/MyTestSuiteGrpc.yml:
```diff
@@ -0,0 +1,12 @@
+        method: "{{.grpcMethod}}"
+        plaintext: true
+        assertions:
+          - result.statuscode ShouldEqual 0
```
```suggestion
          - result.code ShouldEqual 0
```

matoous replied:
Fixed :)

yesnault commented:
@matoous thank you! It's merged on a branch to update vendor/ directory before going on master (with no squash to keep your commit of course)
